### ISSUE TITLE
sql/adminui: upgrade from 19.2.X to 20.2.x breaks Data Distribution page due to missing namespace2 zone config

### ISSUE THREAD

**Issue by rafiss** [cockroachdb/cockroach#49882]
**Describe the problem**

Going to the data distribution page results in an internal server error, and the following appears in the logs:
```
admin-replica-matrix: relation "system.public.namespace2" does not exist.
```

**To Reproduce**

1. Start a 19.2.6 cluster.
2. Upgrade it to a 20.1.0 or 20.1.1 cluster.
3. Go to the Data Distribution page (e.g. http://localhost:8081/#/data-distribution)
4. This page errors.

**Expected behavior**
The page should work normally.

**Additional data / screenshots**
<img width="1404" alt="image" src="https://user-images.githubusercontent.com/1320573/83807227-bd4a2880-a680-11ea-8853-9626780a11be.png">

The problem is happening because this page issues the following query:

```
root@:26257/defaultdb> SELECT name, table_id, database_name, drop_time FROM "".crdb_internal.tables WHERE schema_name = 'public';

               name               | table_id | database_name | drop_time
----------------------------------+----------+---------------+------------
  namespace                       |        2 | system        | NULL
  descriptor                      |        3 | system        | NULL
  users                           |        4 | system        | NULL
  zones                           |        5 | system        | NULL
  settings                        |        6 | system        | NULL
  lease                           |       11 | system        | NULL
  eventlog                        |       12 | system        | NULL
  rangelog                        |       13 | system        | NULL
  ui                              |       14 | system        | NULL
  jobs                            |       15 | system        | NULL
  web_sessions                    |       19 | system        | NULL
  table_statistics                |       20 | system        | NULL
  locations                       |       21 | system        | NULL
  role_members                    |       23 | system        | NULL
  comments                        |       24 | system        | NULL
  replication_constraint_stats    |       25 | system        | NULL
  replication_critical_localities |       26 | system        | NULL
  replication_stats               |       27 | system        | NULL
  reports_meta                    |       28 | system        | NULL
  namespace2                      |       30 | system        | NULL
  protected_ts_meta               |       31 | system        | NULL
  protected_ts_records            |       32 | system        | NULL
  role_options                    |       33 | system        | NULL
  statement_bundle_chunks         |       34 | system        | NULL
  statement_diagnostics_requests  |       35 | system        | NULL
  statement_diagnostics           |       36 | system        | NULL
```

For each row in the above result, it issues the query `SELECT zone_id FROM [SHOW ZONE CONFIGURATION FOR TABLE <table>`. This results in an error for the `namespace2` table.

```
root@:26257/defaultdb> SELECT zone_id FROM [SHOW ZONE CONFIGURATION FOR TABLE system.public.namespace2];
ERROR: relation "system.public.namespace2" does not exist
SQLSTATE: 42P01
```

**Environment:**
 - CockroachDB version 20.1.1

**Comment by ajwerner** [comment]
The crux of the above issue is that `"".crdb_internal.tables` is driven by `(*TableCollection).getAllDescriptors()` which returns descriptors for both the old and new `namespace` table and reads the name from the descriptor itself. The challenge is that there are two tables that we're going to migrate between and we don't know whether the migration has happened unless we go and read a migration key which I'm not sure we want to do.

I see two layers at which we could fix this:

1) Remap the rows of the query in the admin server. I suspect we want to show the data in `namespace2`. To achieve that we could, after running the query, take the descriptor from the `namespace2` row and plop that into the `namespace` row. We could choose to do this remapping if the migration has happened, which I suppose we can cache somewhere. 

2) Prevent `namespace2` from popping out of `getAllDescriptors` in the first place. This feels both better on the one hand, because we are indeed hiding the existence of `namespace` from nodes as soon as it comes into existence. On the other hand, it adds a good bit of complexity. 

My hunch is that the right thing to do is 1). It's a minimally invasive change and even if we do hide the data that exists in namespace pre-migration, who cares, we don't render the `namespace` table in the data distribution page at all as far as I can tell.

**Comment by ajwerner** [comment]
Nevermind, the right fix seems to be a tad simpler. Zone configs use a special lookup path when mapping a name to an ID. I'm not sure why that is but there's a low impact way to make it work correctly so I'll be doing that.

**Comment by ajwerner** [comment]
Closed by  #49987

**Comment by rafiss** [comment]
Reopening this because the same issue happens if you upgrade from v19.2 to v20.1 to v20.2. I found this through a report here: https://cockroachdb.slack.com/archives/CP4D9LD5F/p1616680160130300

**Comment by israellot** [comment]
I was affected by this issue when upgrading from 19.2.x to 20.2.x and now that I'm running 21.1.11 it affects my deployment again. Checking logs I see the same missing namespace2 error.

**Comment by ajwerner** [comment]
Okay, I've got some rather small patches up to help fix this for 20.2 and will put one up for 21.1 as soon as I can get that branch to build. Fortunately it does not affect 21.2, which is due out very soon. 

https://github.com/cockroachdb/cockroach/pull/72506

**Comment by ajwerner** [comment]
Closing with #72506 and #72507.
